projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -162,6 +162,14 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent> {
         }
     }
 
+    public get_row_index_in_data(id: string, rowID: any): number {
+        const grid = this.get(id) as IgxGridBaseComponent;
+        if (!grid) {
+            return -1;
+        }
+        return grid.primaryKey ? grid.data.map((record) => record[grid.primaryKey]).indexOf(rowID) : grid.data.indexOf(rowID);
```

Comment:
Creating a separate array of ID-s lokks like unnecessary overhead. 
Just find the record in question with `.find()`/`.findIndex()`